### PR TITLE
feat: Deep connector Zendesk Support (Metadata)

### PR DESCRIPTION
**Pull request by Cobalt0s**
https://github.com/amp-labs/connectors/pull/631

Closes https://github.com/amp-labs/connectors/issues/616

### CODE REVIEW

**Comment by Cobalt0s** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/722?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#722** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/722?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#716** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/716?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#717](https://github.com/amp-labs/connectors/pull/717) <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/717?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#721](https://github.com/amp-labs/connectors/pull/721) <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/721?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#715** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/715?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#661** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/661?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#670](https://github.com/amp-labs/connectors/pull/670) <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/670?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#720** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/720?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#710](https://github.com/amp-labs/connectors/pull/710) <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/710?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#712** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/712?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#713](https://github.com/amp-labs/connectors/pull/713) <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/713?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#711** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/711?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Cobalt0s and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: test/zendesksupport/metadata/main.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		utils.Fail("error listing metadata for ZendeskSupport", "error", err)
+	}
+
+	fmt.Println("Compare object metadata against endpoint response:")
+
+	mismatchErr := mockutils.ValidateReadConformsMetadata(objectName, response.Data[0].Raw, metadata)
+	if mismatchErr != nil {
+		utils.Fail("schema and payload response have mismatching fields", "error", mismatchErr)
+	} else {
+		fmt.Println("==> success fields match.")
```
Comment (eberle1080) [inline test/zendesksupport/metadata/main.go:56]: I'd use slog here

**Review by eberle1080 — APPROVED** [review]
A few small bits of feedback, plus a question about caching, but no blockers.

File: test/zendesksupport/metadata/main.go (hunk 1 above)
Comment (Cobalt0s) [reply test/zendesksupport/metadata/main.go:56]: Files under scripts and test use fmt. But I avoided using such in main connector packages.
I will update fmt only that are part of this PR. For the rest would be better to have a dedicated PR.

**Pull request merged**